fedimint-atomic-broadcast/src/network.rs

```diff
@@ -0,0 +1,98 @@
+use std::io::Write;
+
+
+use bitcoin_hashes::{sha256, Hash};
+use parity_scale_codec::{Decode, Encode, IoReader};
+
+use crate::conversion::to_peer_id;
+use crate::data_provider::UnitData;
+use crate::keychain::Keychain;
+use crate::{Message, Recipient};
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Hasher;
+
+impl aleph_bft::Hasher for Hasher {
+    type Hash = [u8; 32];
+
+    fn hash(input: &[u8]) -> Self::Hash {
+        let mut engine = sha256::HashEngine::default();
+        engine
+            .write_all(input)
+            .expect("Writing to a hash engine cannot fail");
+
+        sha256::Hash::from_engine(engine).to_byte_array()
+    }
+}
+
+pub type NetworkData = aleph_bft::NetworkData<
+    Hasher,
+    UnitData,
+    <Keychain as aleph_bft::Keychain>::Signature,
+    <Keychain as aleph_bft::MultiKeychain>::PartialMultisignature,
+>;
+
+pub struct Network {
```

Comment:
Going through this PR I'm generally confused about thread&channel architecture of the whole thing. I think it would be worth drawing via mermaid chart in the main `lib.rs`

Anyway - `Network` does not seem like the right name for two inter-task/thread communication endpoints. Not sure how exactly to name it.

Reply:
Oh dear. Now I see it was probably named this way due to `aleph_bft::Network` :shrug:

Reply:
Exactly

Reply:
It is *some* reason, though I don't think we necessarily need to follow it. I think it's OK if "(our) NetworkComm implements alephf Network", etc.